app/uk/gov/hmrc/helpfrontend/controllers/HelpController.scala

```diff
@@ -58,9 +64,26 @@ class HelpController @Inject() (
     Ok(cookiesPage())
   }
 
-  val onlineServicesTerms: Action[AnyContent] = Action { implicit request =>
+  def onlineServicesTerms(lang: Option[String] = None): Action[AnyContent] = Action { implicit request =>
```

Comment:
@kyle-bowden I wonder if we could have something like this, reduces the amount of changes

```
Action { initialRequest => 
     implicit request = Try(initialRequest.withTransientLang(lang.map(Lang(_)).get)).getOrElse(initialRequest)
     
     ... what was there before, maybe just always having .withLang(request.lang) on end of result for readability
}
```

Reply:
or have an action transformer https://www.playframework.com/documentation/2.8.x/ScalaActionsComposition

Reply:
cleaned up my badly broken psuedocode! above!

Reply:
```
def onlineServicesTerms(langFromURL) = maybeOverrideLang(langFromURL) {
    Action { implicit request =>
        ... as before
    }
}
```
or would it have to be like this without the extra class stuff? 🤷‍♂️ 
```
def onlineServicesTerms(langFromURL) = maybeOverrideLang(langFromURL)(
    Action { implicit request =>
        ... as before
    }
)
```

Reply:
i did think about an ActionTransformer but thought it might be over-engineering for this one endpoint.  i really like the look of the above, though, and it takes all the complexity out of the controller.  definitely worth a timeboxed attempt to get it working.